src/vm.nit

```diff
@@ -706,6 +750,19 @@ redef class Instance
 	var vtable: nullable VTable
 end
 
+redef class Variable
+	# The position in the environment
+	var position: Int
+end
+
+# Implementation of a Frame with numbered variables
+class VmFrame
+	super Frame
+
+	# Contains the value of Variables (which are numbered)
+	var environment: Array[Instance] = new Array[Instance]
```

Comment:
Oh. here is the `environment`. Is is still a bad word (the original `map` is a bad word too in fact). what about `variables`  or `locals`?